app/response_svc.py

```diff
@@ -1,16 +1,115 @@
 from aiohttp_jinja2 import template
+import asyncio
+import json
+import uuid
 
+from app.objects.secondclass.c_fact import Fact
+from app.objects.c_operation import Operation
+from app.objects.c_source import Source
 from app.utility.base_service import BaseService
 
 
+BLUE_ADVERSARY = 'f61e3fc0-43d8-4b36-b5d3-710610b92974'
+BLUE_OP_NAME = 'Auto-Collect'
+
+
+async def handle_link_completed(socket, path, services):
+    data = json.loads(await socket.recv())
+    paw = data['agent']['paw']
+    data_svc = services.get('data_svc')
+
+    agent = await data_svc.locate('agents', match=dict(paw=paw, access=data_svc.Access.RED))
+    if agent:
+        pid = data['pid']
+        return await services.get('response_svc').respond_to_pid(pid, agent[0])
+
+
 class ResponseService(BaseService):
 
     def __init__(self, services):
         self.log = self.add_service('response_svc', self)
         self.data_svc = services.get('data_svc')
+        self.rest_svc = services.get('rest_svc')
+        self.agents = []
+        self.adversary = None
+        self.abilities = []
+        self.op = None
 
     @template('response.html')
     async def splash(self, request):
         abilities = [a for a in await self.data_svc.locate('abilities') if await a.which_plugin() == 'response']
         adversaries = [a for a in await self.data_svc.locate('adversaries') if await a.which_plugin() == 'response']
         return dict(abilities=abilities, adversaries=adversaries)
+
+    @staticmethod
+    async def register_handler(event_svc):
+        await event_svc.observe_event('link/completed', handle_link_completed)
+
+    async def respond_to_pid(self, pid, agent):
```

Comment:
can you take a minute to refactor this: it can be broken into smaller easier to read functions